src/test/java/htsjdk/samtools/SAMFileWriterFactoryTest.java

```diff
@@ -304,4 +320,75 @@ public void testAsync() throws IOException {
         writer = builder.setUseAsyncIo(false).makeWriter(header, false, outputFile, referenceFile);
         Assert.assertFalse(writer instanceof AsyncSAMFileWriter, "testAsync option=unset");
     }
+
+    @Test
+    public void testMakeWriterForSamExtension() throws IOException {
+        final File tmpFile = File.createTempFile("testMakeWriterForSamExtension", ".sam");
```

Comment:
Rather than writing ".sam", can you use `SamReader.Type.SAM_TYPE.fileExtension()`.